server/src/main/java/org/opensearch/cluster/node/DiscoveryNodeRole.java

```diff
@@ -349,7 +348,7 @@ public void validateRole(List<DiscoveryNodeRole> roles) {
      * The version that {@link #REMOTE_CLUSTER_CLIENT_ROLE} is introduced. Nodes before this version do not have that role even
      * they can connect to remote clusters.
      */
-    public static final Version REMOTE_CLUSTER_CLIENT_ROLE_VERSION = LegacyESVersion.fromString("7.8.0");
```

Comment:
`REMOTE_CLUSTER_CLIENT_ROLE_VERSION` and `LEGACY_ROLES` (L354) are not used in the code base. Let's remove these variables entirely.